### PR TITLE
Avoid method chaining

### LINKED ISSUE #90: [Quality] avoid method chaining
**Description**
As a developer I want to have an easy to maintain codebase so that changes are do not have a big impact on other implementations. This can be achieved by avoiding method chaining.

**Ready Criteria**
- [ ] everyone understood the scope of this ticket
- [ ] effects in domain- / useCase- / external-layers are discussed
- [ ] atleast one person is assigned

**Acceptance Criteria**
- [ ] Every method got updated
- [ ] Tests work the same as before

**Scope**
✅ Every method should be updated
❌  no further refactoring

**Additional context**
Example:
```java
email.getEmailMetadata().recipients().cc()
```
in the aggregate root it should be changed to 

```java
email.getRecipientCC()
```

### PR DESCRIPTION
**Pull request by seiferla**
fixes #90

### CODE REVIEW

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Email.java [hunk 1]
```diff
@@ -56,4 +61,8 @@ public int hashCode() {
     public EmailMetadata getEmailMetadata() {
         return emailMetadata;
     }
+
+    public List<Address> getRecipientCC() {
+        return this.recipientCC;
+    }
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Email.java:67]:
```suggestion
    public List<Address> getRecipientCC() {
        return this.emailMetadata.recipients().cc();
    }
```
We have no need for a new instance field here.

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/imap/ImapListenerConfig.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/imap/ImapListenerConfig.java:1]: Should not be part of this PR

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java [hunk 2]
```diff
@@ -21,4 +21,8 @@ public record User(Address email, String password, String salt) {
       throw new InvalidSaltException("Salt must not be null or empty");
     }
   }
+
+  public boolean hasEmail(Address email) {
+    return this.email.equals(email);
+  }
 }
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/user/User.java:28]: Nice!

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java [hunk 3]
```diff
@@ -15,11 +16,16 @@ private CreateEmailHelper() {}
 
   public static Email createEmailWithMessage(Message message) throws EmailParsingException {
     try {
-      return Email.create(message.getContent().toString(),
-          EmailMetadataFactory.withMessage(message).build(), null);
-    } catch (IOException | MessagingException e) {
+
+      String content = message.getContentType();
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java:20]:
```suggestion
      String content = message.getContent().toString()
```
Was this by accident or did I miss something?

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java [hunk 4]
```diff
@@ -15,11 +16,16 @@ private CreateEmailHelper() {}
...
+
+      EmailMetadataFactory factory = EmailMetadataFactory.withMessage(message);
+
+      EmailMetadata metadata = factory.build();
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java:24]:
```suggestion
      EmailMetadata factory = EmailMetadataFactory.withMessage(message).build();
```
Typically you write it like this when working with factories

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/james/JamesImapProcessor.java
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/james/JamesImapProcessor.java:1]: Did you change anything or did you just format this file?

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java (hunk 3 above)
Comment (seiferla) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java:20]: sonarlint displayed me that getContentType return already a String

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/james/JamesImapProcessor.java
Comment (seiferla) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/james/JamesImapProcessor.java:1]: I changed the processCapabilityRequest method. This is not really a method chaining issue, but Jens tagged it as a refactor. So I split the code into different methods sendStatusResponse and createCapabilityResponse
and maybe I accidentally formatted this file

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java (hunk 3 above)
Comment (Jonas-Karl) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/email/CreateEmailHelper.java:20]: Yes because the contentType is a string. But previously we wrote getContent instead of getContentType

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/imap/ImapListenerConfig.java
Comment (seiferla) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/imap/ImapListenerConfig.java:1]: do you mean the file or the code?

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Email.java [hunk 5]
```diff
@@ -56,4 +60,28 @@ public int hashCode() {
     public EmailMetadata getEmailMetadata() {
         return emailMetadata;
     }
```
Comment (Jonas-Karl) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Email.java:62]: no need for this getter anymore

**Review by Jonas-Karl — APPROVED** [review]
